File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector <vector <int>> res;
        sort(intervals.begin(), intervals.end(), cmp);
        vector <int> cur = intervals[0];
        for (int i = 1; i < intervals.size(); i++) {
            if (intervals[i][0] > cur[1]) {
                res.push_back(cur);
                cur = intervals[i];
            } else {
                cur[1] = max(cur[1], intervals[i][1]);
            }
        }
        res.push_back(cur);
        return res;
    }
    static bool cmp(vector <int> &x, vector <int> &y) {
        return x[0] < y[0];
    }
};

File: 0056-merge-intervals/0056-merge-intervals.go
func merge(intervals [][]int) [][]int {
    sort.Slice(intervals, func (i, j int) bool {
        return intervals[i][0] < intervals[j][0]
    })
    var (
        last = []int{intervals[0][0], intervals[0][1]}
        ans  = make([][]int, 0)
    )
    for i := 1; i < len(intervals); i++ {
        if intervals[i][0] <= last[1] {
            if last[1] < intervals[i][1] {
                last[1] = intervals[i][1]
            }
        } else {
            ans = append(ans, last)
            last = []int{intervals[i][0], intervals[i][1]}
        }
    }
    ans = append(ans, last)
    return ans
}